tools/hangar/__snapshots__/test_corpus/add_consumer.w_compile_tf-aws.md

```diff
@@ -8,7 +8,7 @@ module.exports = function({ c }) {
     }
     async handle(msg)  {
       {
-        (await c.inc());
+        (typeof c.inc === "function" ? await c.inc() : await c.inc.handle());
```

Comment:
Curious- do we really need a runtime check? don't we know at compile time if this is supposed to be a function or a handler?

Reply:
Perhaps. In an expression like `a.b.c()`, if the last reference component (`c`) is a preflight member of its parent (`a.b`), then we know it's going to be a class instance in JS, while if the last variable component is an inflight member (for example, an inflight method of a class, or an inflight class member with a function type), then it will be plain old JS function. But I'm not sure there isn't some case I'm missing here. In any case I feel like we can leave this as a future optimization

Reply:
Can you check if this is a low hanging fruit? Feels like this test should be easy to make at the jsification stage. No?

Reply:
@eladb After some pondering, here's the counter example showing why we can't [always] statically determine whether it's a function or handler. I'll add it as a unit test:

```js
bring cloud;

class Foo {
  inflight1: inflight (): num;
  init() {
    // here is an inflight function created during preflight
    this.inflight1 = inflight (): num => {
      return 1;
    };
  }

  inflight inflight2: inflight (): num;
  inflight init() {
    // here is an inflight function created during inflight
    this.inflight2 = inflight (): num => {
      return 2;
    };
  }

  inflight makeFn(x: bool): inflight (): num {
    if (x == true) {
      return this.inflight1;
    } else {
      return this.inflight2;
    }
  }

  inflight callFn(x: bool): num {
    // partialFn could be an inflight function created during preflight or inflight,
    // so we have to code-generate this to work for both cases
    let partialFn = this.makeFn(x);
    return partialFn();
  }
}

let foo = new Foo();

test "calling inflight functions created in different phases" {
  assert(foo.callFn(true) == 1);
  assert(foo.callFn(false) == 2);
}
```

Reply:
Thanks for the clarification. It might still be possible to optimize in cases we know for certain, but that's something we can do later on.